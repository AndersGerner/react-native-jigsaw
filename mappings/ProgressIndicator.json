{"name":"Progress Indicator","tag":"ProgressIndicator","description":"A component used to show current Step in a process","category":"formControl","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Control_Slider.png","supports_list_render":false,"layout":{},"props":{"numberOfSteps":{"label":"Number of Steps","description":"The border radius for the thumb","editable":true,"required":false,"type":"number","min":0,"max":10,"step":1,"precision":0,"value":5},"currentStep":{"label":"Current Step","description":"The step that the user is on","editable":true,"required":false,"type":"number","min":0,"max":10,"step":1,"precision":0,"value":5},"stepIndicatorSize":{"label":"Step Size","description":"The size of the step circle","editable":true,"required":false,"type":"number","min":10,"max":100,"step":5,"precision":0,"value":25},"stepIndicatorLabelFontSize":{"label":"Step Font Size","description":"The font size of the step number","editable":true,"required":false,"type":"number","min":10,"max":50,"step":1,"precision":0,"value":15},"currentStepIndicatorLabelFontSize":{"label":"Current Step Font Size","description":"The font size of the current step number","editable":true,"required":false,"type":"number","min":10,"max":50,"step":1,"precision":0,"value":15},"unfinishedColor":{"label":"Unfinished Color","description":"The color of the step and divider when the step is either not the current step or not finished","editable":true,"required":false,"type":"color","value":null},"finishedColor":{"label":"Finished Color","description":"The color of the step and divider when the step is the current step or finished","editable":true,"required":false,"type":"color","value":null},"stepNumberUnfinishedColor":{"label":"Unfinished Number Color","description":"The color of the number within the step when unfinished","editable":true,"required":false,"type":"color","value":null},"stepNumberFinishedColor":{"label":"Finished Number Color","description":"The color of the number within the step when finished","editable":true,"required":false,"type":"color","value":null},"stepIndicatorCurrentColor":{"label":"Current Step Color","description":"The color of current step circle","editable":true,"required":false,"type":"color","value":null},"stepIndicatorLabelCurrentColor":{"label":"Current Step Number Color","description":"The color of current number within the step circle","editable":true,"required":false,"type":"color","value":null},"stepStrokeCurrentColor":{"label":"Current Step Border Color","description":"The color of current border color of the step","editable":true,"required":false,"type":"color","value":null},"currentStepStrokeWidth":{"label":"Current Step Border Width","description":"The width of the border for the current step","editable":true,"required":false,"type":"number","min":1,"max":20,"step":1,"precision":0,"value":0},"currentStepIndicatorSize":{"label":"Current Step Size","description":"The size of the current step circle","editable":true,"required":false,"type":"number","min":10,"max":100,"step":5,"precision":0,"value":null}}}