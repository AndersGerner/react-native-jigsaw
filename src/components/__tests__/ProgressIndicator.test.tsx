import * as React from "react";
import renderer from "react-test-renderer";
import ProgressIndicator from "../ProgressIndicator";

it("renders progress indicator", () => {
  const tree = renderer
    .create(
      <ProgressIndicator
        currentStep={1}
        numberOfSteps={5}
        stepIndicatorSize={10}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
