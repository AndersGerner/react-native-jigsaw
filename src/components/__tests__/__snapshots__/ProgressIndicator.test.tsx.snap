// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders progress indicator 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "space-around",
        },
        Object {
          "flexDirection": "row",
          "height": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(165, 173, 183, 1)",
            "borderColor": "rgba(90, 69, 255, 1)",
            "borderRadius": 5,
            "borderWidth": 3,
            "height": 10,
            "justifyContent": "center",
            "width": 10,
            "zIndex": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(90, 69, 255, 1)",
              "fontSize": 15,
            }
          }
        >
          1
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(165, 173, 183, 1)",
            "borderColor": "rgba(165, 173, 183, 1)",
            "borderRadius": 5,
            "borderWidth": 0,
            "height": 10,
            "justifyContent": "center",
            "width": 10,
            "zIndex": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(90, 69, 255, 1)",
              "fontSize": 12,
              "overflow": "hidden",
            }
          }
        >
          2
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(165, 173, 183, 1)",
            "borderColor": "rgba(165, 173, 183, 1)",
            "borderRadius": 5,
            "borderWidth": 0,
            "height": 10,
            "justifyContent": "center",
            "width": 10,
            "zIndex": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(90, 69, 255, 1)",
              "fontSize": 12,
              "overflow": "hidden",
            }
          }
        >
          3
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(165, 173, 183, 1)",
            "borderColor": "rgba(165, 173, 183, 1)",
            "borderRadius": 5,
            "borderWidth": 0,
            "height": 10,
            "justifyContent": "center",
            "width": 10,
            "zIndex": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(90, 69, 255, 1)",
              "fontSize": 12,
              "overflow": "hidden",
            }
          }
        >
          4
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(165, 173, 183, 1)",
            "borderColor": "rgba(165, 173, 183, 1)",
            "borderRadius": 5,
            "borderWidth": 0,
            "height": 10,
            "justifyContent": "center",
            "width": 10,
            "zIndex": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(90, 69, 255, 1)",
              "fontSize": 12,
              "overflow": "hidden",
            }
          }
        >
          5
        </Text>
      </View>
    </View>
  </View>
</View>
`;
